### PR TITLE
[QSP-9] Provide docs to verify publications/deployments

### PR DESCRIPTION
**Pull request by bbenligiray**
Added docs to allow the users to verify where the Api3Voting app ID in `Api3Template.sol` comes from. Added some more docs to describe the entire DAO+pool configuration. We will actively encourage the users to do this verification themselves.

### CODE REVIEW

**Review by BritikovKI — APPROVED** [review]
Seems legitimate

File: packages/dao/scripts/deriveNamehash.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+const namehash = require("eth-ens-namehash");
```
Comment (BritikovKI) [inline packages/dao/scripts/deriveNamehash.js:1]: The only thing which is strange for me, do we really need this file?

Comment (bbenligiray) [reply packages/dao/scripts/deriveNamehash.js:1]: To call `namehash.hash()`

Comment (bbenligiray) [reply packages/dao/scripts/deriveNamehash.js:1]: Ah sorry, I understood your question now. The issue mentions that they don't know where the app ID comes from so they can't verify it, so they can verify that the app ID is the namehash of the app name (`api3voting.open.aragonpm.eth`). How to use it is in the README.

**Pull request merged**